torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -7164,4 +7165,549 @@ Tensor values_backward(const Tensor& grad, const Tensor& self) {
   return grad_self;
 }
 
+std::tuple<Tensor, Tensor> _betainc_even_partial_numerator(const int32_t iteration, const Tensor& a, const Tensor& b, const Tensor& x) {
+  // Even partial numerator used in the continued fraction for betainc.
+  /*
+   * This function computes the partial numerator d_{2m} that is specified
+   * here: https://dlmf.nist.gov/8.17.E23
+   */
+  auto options = at::TensorOptions().dtype(x.dtype()).device(x.device());
+  const Tensor one = at::scalar_tensor(1.0, options);
+  const Tensor two = at::scalar_tensor(2.0, options);
```

Comment:
wrapping should be done automatically below no? You should be able to use 1. or 2. in the formulas

Reply:
I defined the constant scalar tensors (1., 2., .5, etc) explicitly, to have the same data type and the same device with other operands. The constant tensors can be reused in following formulas.

It's the coding style corresponding with the python codes of tensorflow-probability.
I guess the tf-probability tried to avoid wrapping python number objects dynamically and repeatedly, for performance.
I think that the c++ codes of PyTorch can be different from the python codes of tf-probability.
For example, the tensors wrapping 'rvalue constant scalar values' might be reused automatically in compile time.

If you think the definitions aren't helpful for performance, I'll change the definitions to rvalue constants like 1. or 2.

Reply:
We don't really wrap numbers into Tensors using our "Scalar" overloads! So these are just kept as full precision numbers and avoid using Tensors altogether.